### PR TITLE
130 remove deprecated dependencies

### LINKED ISSUE #130: Remove deprecated dependencies
Clean up some deprecated dependencies from ChronoLog directory tree
1. hcldependency removal
2. socketpp & rockdb removal
3. deprecated files removal. Remove them from the makefiles that they are referenced in.

### PR DESCRIPTION
**Pull request by fkengun**
- HCL, socketpp and RocksDB dependencies are removed
- Deprecated files are removed, CMakeLists.txt files are updated accordingly
- Unused test cases are removed
- CMakeLists.txt files are cleaned up
- Only one ```default_conf.json.in``` is all we need for configuration file

### CODE REVIEW

File: CI/enviroment/spack.lock [hunk 1]
```diff
@@ -16,10 +16,6 @@
     {
```
Comment (ibrodkin) [inline CI/enviroment/spack.lock:16]: I believe spack.lock file is create by "spack env activate" and removed by "spack env deactivate". Is there a reason we have in git repo?

File: ChronoKeeper/CMakeLists.txt [hunk 2]
```diff
@@ -1,7 +1,3 @@
-cmake_minimum_required(VERSION 3.19)
-
```
Comment (ibrodkin) [inline ChronoKeeper/CMakeLists.txt:2]: cmake_minimum_required(..)  and project()  are required by cmake , are they not?  Is there a reason to remove them?

File: ChronoStore/CMakeLists.txt [hunk 3]
```diff
@@ -1,14 +1,6 @@
-cmake_minimum_required(VERSION 3.17)
-project(chronostore)
-
 find_package(HDF5 REQUIRED)
 find_package(JSON-C REQUIRED)
```
Comment (ibrodkin) [inline ChronoStore/CMakeLists.txt:6]: json-c needs to be added to spack.yaml

File: Client/CMakeLists.txt [hunk 4]
```diff
@@ -1,9 +1,3 @@
-project(ChronoLogClient)
-
-set(CMAKE_CXX_STANDARD 17)
-
-find_package(thallium REQUIRED)
```
Comment (ibrodkin) [inline Client/CMakeLists.txt:5]: what is the reason find_package(thallium REQUIRED) is removed from Client, ChronoKeeper, and ChronoVisor CMakeLists ?

File: CI/enviroment/spack.lock (hunk 1 above)
Comment (fkengun) [reply CI/enviroment/spack.lock:16]: > I believe spack.lock file is create by "spack env activate" and removed by "spack env deactivate". Is there a reason we have in git repo?

yes, ```spack.lock``` should be removed from the repo.

File: ChronoKeeper/CMakeLists.txt (hunk 2 above)
Comment (fkengun) [reply ChronoKeeper/CMakeLists.txt:2]: > cmake_minimum_required(..) and project() are required by cmake , are they not? Is there a reason to remove them?

it is required if you only want to load the ```CMakeLists.txt``` for the ChronoKeeper. I assume everyone loads the entire project using the root ```CMakeLists.txt```. then this one is not needed.

File: ChronoStore/CMakeLists.txt (hunk 3 above)
Comment (fkengun) [reply ChronoStore/CMakeLists.txt:6]: > json-c needs to be added to spack.yaml

I believe ```JSON-C``` is a dependency of ```Thallium```. so if my assumption that this repo is loaded as a whole stands, that dependency is implied already.

File: Client/CMakeLists.txt (hunk 4 above)
Comment (fkengun) [reply Client/CMakeLists.txt:5]: > what is the reason find_package(thallium REQUIRED) is removed from Client, ChronoKeeper, and ChronoVisor CMakeLists ?

same as above, all of these are based on the same assumption.

**Review by ibrodkin — APPROVED** [review]

**Review by kmk142789 — APPROVED** [review]

**Pull request merged**